libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -617,3 +618,44 @@ void AP_Baro::set_pressure_correction(uint8_t instance, float p_correction)
         sensors[instance].p_correction = p_correction;
     }
 }
+
+/*
+ * register new baro if there is space for it
+ */
+bool AP_Baro::register_uavcan_baro(uint8_t mgr, uint8_t node)
+{
+#if HAL_WITH_UAVCAN
+    // Check if this node in network is already registered
+    for (uint8_t i = 0; i < _num_drivers; i++) {
+        if (sensors[i].can_manager == mgr && sensors[i].uavcan_node == node) {
+            return true;
+        }
+    }
+
+    // If there is place left - allocate one UAVCAN based baro
+    if (_num_drivers < BARO_MAX_DRIVERS && _num_sensors < BARO_MAX_INSTANCES)
+    {
+        if ((AP_BoardConfig_CAN::get_can_enable() != 0) && (hal.can_mgr[mgr] != nullptr))
+        {
+            drivers[_num_drivers] = new AP_Baro_UAVCAN(*this);
+
+            if (((AP_Baro_UAVCAN *) drivers[_num_drivers])->register_uavcan_baro(mgr, node)) {
```

Comment:
Decision on whether to create the object will still be in AP_Baro as the only class that know about all instances. And all the work with registration is already in AP_Baro_UAVCAN::register_uavcan_baro.
However what can be done is to eliminate manager and node array variables in AP_Baro and use the stored values in AP_Baro_UAVCAN!